# Sectoral $CO_2$ emissions by country and year

## About the dataset
The CO2 Emissions Dataset offers comprehensive information on CO2 emissions across countries and sectors, aiding researchers and policymakers in understanding and addressing their impact on the environment.

In [2]:
# Import libraries
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [6]:
file_path = os.getcwd() + "/dataset.csv"

In [19]:
# load dataset
emissions = pd.read_csv(file_path)
print(emissions.head())

     country        date sector      value   timestamp
0     Brazil  01/01/2019  Power   0.096799  1546300800
1      China  01/01/2019  Power  14.816100  1546300800
2  EU27 & UK  01/01/2019  Power   1.886490  1546300800
3     France  01/01/2019  Power   0.051217  1546300800
4    Germany  01/01/2019  Power   0.315002  1546300800


In [20]:
emissions.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 135408 entries, 0 to 135407
Data columns (total 5 columns):
 #   Column     Non-Null Count   Dtype  
---  ------     --------------   -----  
 0   country    135408 non-null  object 
 1   date       135408 non-null  object 
 2   sector     135408 non-null  object 
 3   value      135408 non-null  float64
 4   timestamp  135408 non-null  int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 5.2+ MB


In [26]:
print(emissions['country'].value_counts())

country
Brazil       9672
China        9672
EU27 & UK    9672
France       9672
Germany      9672
India        9672
Italy        9672
Japan        9672
ROW          9672
Russia       9672
Spain        9672
UK           9672
US           9672
WORLD        9672
Name: count, dtype: int64


In [27]:
print(emissions['sector'].value_counts())

sector
Power                     22568
Industry                  22568
Ground Transport          22568
Residential               22568
Domestic Aviation         22568
International Aviation    22568
Name: count, dtype: int64


In [35]:
print(emissions['date'].unique())

['01/01/2019' '02/01/2019' '03/01/2019' ... '29/05/2023' '30/05/2023'
 '31/05/2023']


For each country/region there is a total of 9672 observations, corresponding to information for each of the 6 sectos over the interval of 1612 days - from 01/01/2019 to 31/05/2023.